ITR/data/base_providers.py

```diff
@@ -365,10 +382,10 @@ def get_company_fundamentals(self, company_ids: List[str]) -> pd.DataFrame:
         :param company_ids: A list of company IDs
         :return: A pandas DataFrame with company fundamental info per company (company_id is a column)
         """
+        excluded_cols = ['projected_targets', 'projected_intensities', 'historic_data']
         return pd.DataFrame.from_records(
-            [ICompanyData.parse_obj(c.dict()).dict() for c in self.get_company_data(company_ids)],
-            exclude=['projected_targets', 'projected_intensities', 'historic_data']).set_index(
-                self.column_config.COMPANY_ID)
+            [ICompanyData.parse_obj({k:v for k, v in c.dict().items() if k not in excluded_cols}).dict()
+             for c in self.get_company_data(company_ids)]).set_index(self.column_config.COMPANY_ID)
```

Comment:
Previously we parsed the entire object, only to throw away these three items after parsing them.  If each was a simply string, it would be a small optimization.  But as projections are data from 2019-2050, constructing and then discarding three 40+ element lists is just wasteful.  So we discard the elements before constructing the object to be parsed into a model.